create-nex-server/src/create-nex-server.ts

```diff
@@ -0,0 +1,246 @@
+	.action((name) => {
+		fullName = name?.trim();
+	})
+	.option('-n, --name <name>', 'Server (full) name')
+	.option('-N, --short-name <name>', 'Server (short) name')
+	.option('-v, --nex-version <version>', 'Server NEX version')
+	.option('-k, --access-key <key>', 'Server access key')
+	.option('-s, --session-key-size <size>', 'Size of the session key inside the Kerberos ticket', parseInt)
+	.option('-f, --fragment-size <size>', 'Size of DATA packet fragments', parseInt)
+	.parse(process.argv)
+	.opts();
+
+async function run() {
+	if (program.name) {
+		fullName = program.name;
+	}
+
+	if (!fullName) {
+		const response = await prompts({
+			type: 'text',
+			name: 'path',
+			message: 'Server (full) name (e.g., Mario Kart 7):'
+		}, promptOptions);
+
+		if (typeof response.path === 'string') {
+			fullName = response.path.trim();
+		}
+	}
+
+	if (!fullName) {
+		console.log(
+			'\nPlease specify the server name:\n' +
+			`    ${command} ${green('<server-name>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('friends')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.shortName) {
+		console.log(yellow('No short name set. Deriving from full name.'), 'To specify a short name provide', green('--short-name=<name>'));
+		shortName = fullName;
+	}
+
+	if (!program.nexVersion) {
+		let typedValue = ''; // * Allow the user to type in a custom version not in the list
+		const response = await prompts({
+			type: 'autocomplete',
+			name: 'version',
+			message: 'NEX version:',
+			choices: NEX_VERSIONS.map(version => ({ title: version, value: version })),
+			onRender() {
+				typedValue = (this as any).input; // TODO - Augment the prompts type defs
+			}
+		}, promptOptions);
+
+		if (typeof response.version === 'string') {
+			program.nexVersion = response.version.trim();
+		} else {
+			typedValue = typedValue.trim();
+
+			if (semver.valid(typedValue)) {
+				program.nexVersion = typedValue;
+			}
+		}
+	}
+
+	if (!program.nexVersion) {
+		console.log(
+			'\nPlease specify a valid (semver) NEX version:\n' +
+			`    ${command} ${green('--nex-version=<version>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('--nex-version=3.10.0')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.accessKey) {
+		const response = await prompts({
+			type: 'text',
+			name: 'key',
+			message: 'Server access key (e.g, ridfebb9):'
+		}, promptOptions);
+
+		if (typeof response.key === 'string') {
+			program.accessKey = response.key.trim();
+		}
+	}
+
+	if (!program.accessKey) {
+		console.log(
+			'\nPlease specify a server access key:\n' +
+			`    ${command} ${green('--access-key=<key>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('--access-key=ridfebb9')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.sessionKeySize) {
```

Comment:
Since this is for us and for NEX servers, I don't see the need to specify a custom session key size? This could be defined internally based on the NEX version

Reply:
I had it configurable due to the friends server having a different key size, so I assumed there may be other RDV servers with different key sizes

Reply:
*All* NEX 1 servers use session key size 16, so we can just check `nexVersion < 2.0.0` and leave it as 32 otherwise

Reply:
Is this true for Quazal games as well? Or are we not targeting those games with this tool